scripts/buildWebtypes.js

```diff
@@ -0,0 +1,235 @@
+const path = require('path');
+const fs = require('fs');
+const os = require('os');
+const { execSync } = require('child_process');
+
+const PLAIN_WEB_TYPES_FILE = 'web-types.json';
+const LIT_WEB_TYPES_FILE = 'web-types.lit.json';
+
+const blacklistedPackages = [
+  /^vaadin-/,
+  /^component-base/,
+  /^field-base/,
+  /^field-highlighter/,
+  /^icons/,
+  /^input-container/,
+  /^lit-renderer/,
+  /^polymer-legacy-adapter/,
+];
+
+// Additional attributes that will be added to all components
+const additionalAttributes = [
+  // Theme attribute is not properly declared at the moment
+  {
+    name: 'theme',
+    description: 'The theme variants to apply to the component.',
+    type: 'string | null | undefined',
+  },
+];
+
+/**
+ * Get packages using lerna, excluding blacklisted packages
+ */
+function getRelevantPackages() {
+  const output = execSync('./node_modules/.bin/lerna ls --json --loglevel silent');
+  const allPackages = JSON.parse(output.toString()).map((project) => project.name.replace('@vaadin/', ''));
+  return allPackages.filter((pkg) => !blacklistedPackages.some((blacklistedPackage) => pkg.match(blacklistedPackage)));
+}
+
+function loadAnalysis() {
+  const analysisPath = path.resolve('./analysis.json');
+  try {
+    return JSON.parse(fs.readFileSync(analysisPath, 'utf8'));
+  } catch (e) {
+    throw new Error(
+      `Could not read output of the Polymer Analyzer from: ${analysisPath}. Make sure to run the Polymer Analyzer before generating web-types.`,
+    );
+  }
+}
+
+/**
+ * Maps a type string such as `'boolean | null | undefined'` into an array of type
+ * strings, such as `['boolean', 'null', 'undefined']`
+ * @param typeString
+ * @returns {string[]}
+ */
+function mapType(typeString) {
+  const sanitizedTypeString = (typeString || '').replace(/[!()]/g, '');
+  const types = sanitizedTypeString.split('|');
+  return types.map((type) => type.trim());
+}
+
+function createPlainElementDefinition(elementAnalysis) {
+  const attributes = [...elementAnalysis.attributes, ...additionalAttributes].map((attribute) => ({
+    name: attribute.name,
+    description: attribute.description,
+    value: {
+      type: mapType(attribute.type),
+    },
+  }));
+  const properties = elementAnalysis.properties
+    .filter((prop) => prop.privacy === 'public')
+    .filter((prop) => !prop.metadata.polymer.readOnly)
+    .map((prop) => ({
+      name: prop.name,
+      description: prop.description,
+      value: {
+        type: mapType(prop.type),
+      },
+    }));
+  const events = elementAnalysis.events.map((event) => ({
+    name: event.name,
+    description: event.description,
+  }));
+
+  return {
+    name: elementAnalysis.tagname,
+    description: elementAnalysis.description,
+    attributes,
+    js: {
+      properties,
+      events,
+    },
```

Comment:
While this should correctly define properties and events for the element, I have not been able to figure out where IntelliJ would use these. Creating an element from a script using `document.createElement` for example does not make use of the web-types, and just returns a plain element type.